src/components/AlbumSongsTable/AlbumSongsTable.js

```diff
@@ -0,0 +1,110 @@
+import albumSongsTable from "./AlbumSongsTable.html";
+import { MusicService } from "../../services/MusicService";
+import { SongsTableComponent } from "../SongsTable/SongsTable";
+
+export class AlbumsSongTableComponent {
+  constructor(mountPoint, props = {}) {
+    this.mountPoint = mountPoint;
+    this.props = props;
+
+    this.state = {
+      title: null,
+      authors: null,
+      imageURL: null
+    };
+    this.songs = [];
+  }
+
+  querySelectors() {
+    this.albumSongsTable = this.mountPoint.querySelector(
+      ".album-songs-table__container"
+    );
+  }
+
+  fetchAuthors(authorsIds) {
+    return Promise.all(
+      authorsIds.map(authorId => MusicService.getAuthorById(authorId))
+    );
+  }
+
+  fetchInfoBySong(song) {
+    const albumPromise = MusicService.getAlbumById(song.albumId);
+    const authorsPromise = this.fetchAuthors(song.authors);
+    return Promise.all([albumPromise, authorsPromise]);
+  }
+
+  fetchSongs(albumId) {
+    MusicService.getAlbumSongs(albumId)
+      .then(songs => {
+        this.songs = songs;
+
+        return Promise.all(this.songs.map(song => this.fetchInfoBySong(song)));
+      })
+      .then(songsInfo => {
+        songsInfo.forEach((item, i) => {
+          const [album, authorsInfo] = item;
+          this.state.imageURL = album.imageURL;
+          this.state.title = album.name;
+          this.songs[i].album = album;
+          this.songs[i].authorsInfo = authorsInfo;
+        });
+        this.mount({ shouldFetchData: false });
+      });
+  }
+
+  fetchAuthorForTitle(albumId) {
+    Promise.all([MusicService.getAlbums(), MusicService.getAuthors()]).then(
+      ([albums, authors]) => {
+        albums.forEach(album => {
+          if (album.id === albumId) {
+            this.state.authors = album.authors
+              .map(author => this.getArtistNameById(authors, author))
+              .join(", ");
+          }
+        });
+      }
+    );
+  }
+
+  getArtistNameById(authors, id) {
+    return authors.find(author => author.id === id).name;
+  }
+
+  changeStateSong(songId, isPlaying) {
+    this.playingSongId = isPlaying ? songId : null;
+    if (this.mountPoint.querySelector(".album-songs-table__container")) {
+      this.albumSongs.changeStateSong(songId, isPlaying);
+    }
+  }
+
+  mountChildren() {
+    this.albumSongs = new SongsTableComponent(this.albumSongsTable, {
+      data: this.songs,
+      onSongPlay: this.props.onSongPlay,
+      onSongStop: this.props.onSongStop,
+      playingSongId: this.playingSongId
+    });
+    this.albumSongs.mount();
+  }
+
+  mount(params = {}) {
+    const mountingParameters = {
```

Comment:
wat?? you can differentiate from albumId. If albumId was passed - then fetch data, if not - then dont